gemrb/tests/INIImporter/Test_INIImporter.cpp

```diff
@@ -0,0 +1,132 @@
+		this->stream = new FileStream{};
+
+		stream->Open(SAMPLE_FILE);
+		assert(unit.Open(stream));
+		// INIImporter deletes stream internally after use
+		this->stream = nullptr;
+	}
+};
+
+TEST_F(INIImporter_Test, GetTagsCount) {
+	EXPECT_EQ(unit.GetTagsCount(), 4);
+}
+
+TEST_F(INIImporter_Test, GetTagNameByIndex) {
+	EXPECT_EQ(StringView{"sectionA"}, unit.GetTagNameByIndex(0));
+	EXPECT_EQ(StringView{"sectionB"}, unit.GetTagNameByIndex(1));
+	EXPECT_EQ(StringView{"sectionC"}, unit.GetTagNameByIndex(2));
+	EXPECT_EQ(StringView{"sectionC"}, unit.GetTagNameByIndex(3));
+
+	EXPECT_EQ(StringView{}, unit.GetTagNameByIndex(200));
+}
+
+TEST_F(INIImporter_Test, GetKeysCount) {
+	EXPECT_EQ(unit.GetKeysCount(StringView{"SectionA"}), 6);
+	EXPECT_EQ(unit.GetKeysCount(StringView{"sectiona"}), 6);
+	EXPECT_EQ(unit.GetKeysCount(StringView{"SectionB"}), 1);
+	// first tag match
+	EXPECT_EQ(unit.GetKeysCount(StringView{"SectionC"}), 1);
+
+	EXPECT_EQ(unit.GetKeysCount(StringView{"nope"}), 0);
+}
+
+TEST_F(INIImporter_Test, GetKeyNameByIndex) {
+	auto sectionA = StringView{"SectionA"};
+	auto sectionB = StringView{"SectionB"};
+	auto sectionC = StringView{"SectionC"};
+
+	EXPECT_EQ(StringView{"stringValue"}, (unit.GetKeyNameByIndex(sectionA, 0)));
+	EXPECT_EQ(StringView{"intValue"}, (unit.GetKeyNameByIndex(sectionA, 1)));
+	EXPECT_EQ(StringView{"intValue"}, (unit.GetKeyNameByIndex(sectionA, 2)));
+	EXPECT_EQ(StringView{"floatValue"}, (unit.GetKeyNameByIndex(sectionA, 3)));
+	EXPECT_EQ(StringView{"trueValue"}, (unit.GetKeyNameByIndex(sectionA, 4)));
+	EXPECT_EQ(StringView{"anotherStringValue"}, (unit.GetKeyNameByIndex(sectionB, 0)));
+	EXPECT_EQ(StringView{"iAm"}, (unit.GetKeyNameByIndex(sectionC, 0)));
+
+	EXPECT_EQ(StringView{}, (unit.GetKeyNameByIndex(sectionA, 100)));
+	EXPECT_EQ(StringView{}, (unit.GetKeyNameByIndex(StringView{"section404"}, 0)));
+}
+
+TEST_F(INIImporter_Test, GetKeyAsString) {
+	auto sectionA = StringView{"SectionA"};
+
+	EXPECT_EQ(StringView{"abc"}, (unit.GetKeyAsString(sectionA, StringView{"stringValue"})));
+	EXPECT_EQ(StringView{"abc"}, (unit.GetKeyAsString(sectionA, StringView{"Stringvalue"})));
+	EXPECT_EQ(StringView{"n/a"}, (unit.GetKeyAsString(sectionA, StringView{"strongValue"}, StringView{"n/a"})));
+	// first match
+	EXPECT_EQ(StringView{"123"}, (unit.GetKeyAsString(sectionA, StringView{"intValue"})));
+	EXPECT_EQ(StringView{"2.5"}, (unit.GetKeyAsString(sectionA, StringView{"floatValue"})));
+	EXPECT_EQ(StringView{"true"}, (unit.GetKeyAsString(sectionA, StringView{"trueValue"})));
+
+	EXPECT_EQ(StringView{"xyz"}, (unit.GetKeyAsString(StringView{"sectionB"}, StringView{"anotherStringValue"})));
+}
+
+TEST_F(INIImporter_Test, GetKeyAsInt) {
+	auto sectionA = StringView{"SectionA"};
+
+	EXPECT_EQ(123, (unit.GetKeyAsInt(sectionA, StringView{"intValue"}, 0)));
+	EXPECT_EQ(123, (unit.GetKeyAsInt(sectionA, StringView{"Intvalue"}, 0)));
+	EXPECT_EQ(456, (unit.GetKeyAsInt(sectionA, StringView{"intValueX"}, 456)));
```

Comment:
Why not consider this a bug? The key is not present, so one should be able to get a consistent result, not something else in case a similar key exists.